cli/2024-10-cli-udt.md

```diff
@@ -0,0 +1,232 @@
+
+    ```bash
+    rad resource-type delete Mycompany.Messaging/plaidResource
+    Warning: `myapp` application has resources of type `Mycompany.Messaging/plaidResource` provisioned and running. Are you sure you want to delete the resource type plaidResource from the resource provider Mycompany.Messaging? (yes/no)
+    ```
+    The resource type `Plaid` is deleted from the resource provider `Mycompany.Messaging`
+
+8. Deb deletes the resource provider Mycompany.Messaging
+
+    ```bash
+    rad resource-provider delete Mycompany.Messaging
+    Warning: `myapp` application has resources of type `Mycompany.Messaging/plaidResource` provisioned and running. Are you sure you want to delete the resource provider Mycompany.Messaging? (yes/no)
+    ```
+
+## Finalized set of commands
+
+### rad resource-provider
+
+#### 1. rad resource-provider create
+
+Resource providers are entities that implement a group of resource types such as `Applications.Core`. Creating a resource provider involves creating new resource types that can be used in your applications. Inputs can be passed in as a file or as flags.
+
+Create a resource provider from a file 
+    
+    ```bash
+    rad resource-provider create -f Mycompany.Messaging.yaml
+    Resource provider Mycompany.Messaging created successfully
+    ```
+**Flags**
+- `-f`, `--fom-file` : The file that contains the resource provider definition
+
+#### 2. rad resource-provider list
+Resource providers are entities that implement a group of resource types such as `Applications.Core`. Users can list all the resource providers available in your tenant.
+
+    ```bash
+    rad resource-provider list 
+    RESOURCE-PROVIDER   TYPE                                STATE
+    Mycompany.Messaging System.Resources/resourceProviders  Succeeded
+    ```
+
+#### 3. rad resource-provider show
+Resource providers are entities that implement a group of resource types such as `Applications.Core`. User can display the details of the resource providers.
+
+    ```bash
+    rad resource-provider show Mycompany.Messaging
+    RESOURCE-PROVIDER   TYPE                                STATE
+    Mycompany.Messaging System.Resources/resourceProviders  Succeeded
+    ```
+
+#### 4. rad resource-provider delete
+Resource providers are entities that implement a group of resource types such as `Applications.Core`. Users can delete a resource provider.
+
+    ```bash
+    rad resource-provider delete Mycompany.Messaging
+    ```
+**Behaviors**
+Option 1 : Radius can prompt the user with a warning message if there are resources provisioned and running under the resource provider. If the user confirms deletion, Radius cleans up everything, deletes resources, types and resource providers.
+
+Option 2 : Radius can error out if there are resources provisioned and running under the resource provider. The user needs to delete the resources before deleting the resource provider.
+
```

Comment:
@Reshrahim and I have had a few discussions about this. We should nail down both:

- The behavior of the API: The API has to enforce invariants.
- The behavior of the CLI: The CLI can provide convenience for the user.

These don't have to be 100% the same, but they have to be largely in agreement. 

Any behaviors that we want to consistent for all clients, should probably be enforced at the API layer.

Reply:
For deleting a resource type or resource provider, we decided to do the following :
    1. API - Error out of if there are resources provisioned with resource type 
    2. CLI - Prompt the user with a warning message that there are resources provisioned with the resource type. If the user confirms deletion, API behavior is exercised.